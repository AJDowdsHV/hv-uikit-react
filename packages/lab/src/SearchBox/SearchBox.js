/*  TODO: Review accessibility */

/*  eslint-disable  jsx-a11y/click-events-have-key-events */
/*  eslint-disable jsx-a11y/no-static-element-interactions */

import React from "react";
import PropTypes from "prop-types";

import SearchIcon from "@hv/uikit-react-icons/dist/Generic/Search";
import CloseIcon from "@hv/uikit-react-icons/dist/Generic/Close";

class HvSearchBox extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: props.value
    };
  }

  /**
   * Updates the state and fires callback
   */
  changeInputValue = value => {
    const { onChange, onValidate } = this.props;
    if (onValidate && !onValidate(value)) return;
    this.setState({ value });
    onChange(value);
  };

  /**
   * Handles the change of the input
   *
   * @param {Object} event - The event that triggered the action
   */
  handleInputChange = event => {
    const currentValue = event.target.value;
    this.changeInputValue(currentValue);
  };

  /**
   * Handles the clear of the input
   */
  handleInputClear = () => {
    this.changeInputValue("");
  };

  render() {
    const { placeholder, classes, searchInput } = this.props;
    const { value } = this.state;
    const inputValue = searchInput !== undefined ? searchInput : value;

    const hasInputValue = inputValue && inputValue !== "";

    const iconRenderer = hasInputValue ? (
      <span onClick={this.handleInputClear}>
        <CloseIcon className={classes.iconClear} />
      </span>
    ) : (
      <SearchIcon className={classes.icon} />
    );

    return (
      <div
        className={
          hasInputValue ? classes.rootWithInput : classes.rootWithoutInput
        }
      >
        <input
          className={classes.input}
          value={inputValue}
          placeholder={placeholder}
          type="text"
          onChange={this.handleInputChange}
        />
        {iconRenderer}
      </div>
    );
  }
}

HvSearchBox.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied to the search box.
   */
  classes: PropTypes.instanceOf(Object).isRequired,
  /**
   * The initial value of the input
   */
  value: PropTypes.string,
  /**
   * The placeholder value of the input
   */
  placeholder: PropTypes.string,
  /**
   * The function that will be executed when the input changes,
   * it receives the input value
   */
  onChange: PropTypes.func,
  /**
   * The function that use to validate input,
   */
  onValidate: PropTypes.func,
  /**
   * The initial value to be used by the input
   */
  searchInput: PropTypes.string
};

HvSearchBox.defaultProps = {
  value: "",
  placeholder: "Search",
  searchInput: undefined,
  onChange: () => {},
  onValidate: undefined
};

export default HvSearchBox;
