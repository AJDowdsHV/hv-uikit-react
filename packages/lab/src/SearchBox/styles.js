const root = theme => ({
  position: "relative",
  background: theme.hv.palette.atmosphere.atmo1,
  height: "32px",
  paddingLeft: `${theme.hv.spacing.xs}px`,
  paddingRight: `${theme.hv.spacing.md}px`
});

const icon = {
  position: "absolute",
  top: 0,
  right: 0,
  width: "30px",
  height: "30px"
};

const styles = theme => ({
  rootWithoutInput: {
    ...root(theme),
    border: `1px solid ${theme.hv.palette.atmosphere.atmo6}`
  },
  rootWithInput: {
    ...root(theme),
    border: `1px solid ${theme.hv.palette.accent.acce1}`
  },

  input: {
    border: "none",
    height: "30px",
    width: "100%",
    background: "transparent",
    ...theme.hv.typography.normalText,
    "&:focus": {
      outline: "none"
    },
    "&::placeholder": {
      ...theme.hv.typography.normalText,
      color: theme.hv.typography.disabledText.color
    },
    "&::-ms-clear": {
      display: "none"
    }
  },
  icon: {
    ...icon
  },
  iconClear: {
    ...icon,
    cursor: "pointer"
  }
});

export default styles;
