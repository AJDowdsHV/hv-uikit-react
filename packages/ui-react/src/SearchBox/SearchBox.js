/**
 * Copyright (c) 2019 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

import React from "react";
import PropTypes from "prop-types";

import SearchIcon from "@hv-ui/icons/core/S-icons/Search16";
import CloseIcon from "@hv-ui/icons/core/S-icons/Close16";

class HvSearchBox extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: props.value
    };
  }

  /**
   * Updates the state and fires callback
   */
  changeInputValue = value => {
    const { onChange } = this.props;
    this.setState({ value });
    onChange(value);
  };

  /**
   * Handles the change of the input
   *
   * @param {Object} event - The event that triggered the action
   */
  handleInputChange = event => {
    const currentValue = event.target.value;
    this.changeInputValue(currentValue);
  };

  /**
   * Handles the clear of the input
   */
  handleInputClear = () => {
    this.changeInputValue("");
  };

  render() {
    const { placeholder, classes } = this.props;
    const { value } = this.state;

    const hasInputValue = value && value != "";

    const iconRenderer = hasInputValue ? (
      <span onClick={this.handleInputClear}>
        <CloseIcon className={classes.iconClear} />
      </span>
    ) : (
      <SearchIcon className={classes.icon} />
    );

    return (
      <div
        className={
          hasInputValue ? classes.rootWithInput : classes.rootWithoutInput
        }
      >
        <input
          className={classes.input}
          value={value}
          placeholder={placeholder}
          type="text"
          onChange={this.handleInputChange}
        />
        {iconRenderer}
      </div>
    );
  }
}

HvSearchBox.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied to the search box.
   */
  classes: PropTypes.instanceOf(Object).isRequired,
  /**
   * The initial value of the input
   */
  value: PropTypes.string,
  /**
   * The placeholder value of the input
   */
  placeholder: PropTypes.string,
  /**
   * The function that will be executed when the input changes,
   * it receives the input value
   */
  onChange: PropTypes.func
};

HvSearchBox.defaultProps = {
  value: "",
  placeholder: "Search",
  onChange: () => {}
};

export default HvSearchBox;
