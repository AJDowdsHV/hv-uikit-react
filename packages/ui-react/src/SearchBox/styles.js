/**
 * Copyright (c) 2019 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

const icon = {
  position: "absolute",
  right: 0,
  width: "32px",
  height: "32px"
};

const styles = theme => ({
  root: {
    position: "relative",
    background: theme.palette.grey.foggy,
    borderBottom: `1px solid ${theme.palette.grey.plain}`,
    height: "32px",
    paddingLeft: `${theme.spacing.xs}px`,
    paddingRight: `${theme.spacing.md}px`
  },
  input: {
    border: "none",
    height: "32px",
    width: "100%",
    background: "transparent",
    ...theme.typography.body1,
    "&:focus": {
      outline: "none"
    },
    "&::placeholder": {
      ...theme.typography.disabled
    }
  },
  icon: {
    ...icon
  },
  iconClear: {
    ...icon,
    cursor: "pointer"
  }
});

export default styles;
