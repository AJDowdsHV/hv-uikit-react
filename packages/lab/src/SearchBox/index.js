import withStyles from "@material-ui/core/styles/withStyles";
import withDeprecate from "@hv/uikit-react-core/dist/withDeprecate";
import styles from "./styles";
import SearchBox from "./SearchBox";

export default withDeprecate(
  withStyles(styles)(SearchBox),
  "This component is deprecated. Please use the Tooltip Component in the Core Package"
);
