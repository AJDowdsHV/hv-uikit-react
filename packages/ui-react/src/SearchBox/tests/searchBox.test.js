/**
 * Copyright (c) 2019 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

/* eslint-env jest */

import React from "react";
import { shallow, mount } from "enzyme";
import SearchBox from "../SearchBox";

const mockClasses = {};

const testState = (instance, value) => {
  expect(instance.state.value).toEqual(value);
};

describe("<SearchBox />", () => {
  const onChangeMock = jest.fn();

  it("has default props", () => {
    const searchBox = mount(
      <SearchBox classes={mockClasses} onChange={onChangeMock} />
    );
    const props = searchBox.props();
    expect(props.value).toEqual("");
    expect(props.placeholder).toEqual("Search");
    expect(props.onChange).toBe(onChangeMock);
  });

  it("renders correctly", () => {
    const searchBox = shallow(
      <SearchBox classes={mockClasses} onChange={onChangeMock} />
    );
    expect(searchBox).toMatchSnapshot();
  });

  it("onChange is triggered and state is changed", () => {
    onChangeMock.mockReset();
    const searchBox = shallow(
      <SearchBox classes={mockClasses} onChange={onChangeMock} />
    );
    const eventMock = {
      target: {
        value: "test"
      }
    };
    searchBox.find("input").simulate("change", eventMock);

    expect(onChangeMock).toBeCalledWith("test");
    testState(searchBox.instance(), "test");
  });

  it("onChange is triggered and state resets when clear is clicked", () => {
    onChangeMock.mockReset();
    const searchBox = shallow(
      <SearchBox
        classes={mockClasses}
        onChange={onChangeMock}
        value={"mockValue"}
      />
    );
    searchBox.find("span").simulate("click");

    expect(onChangeMock).toBeCalledWith("");
    testState(searchBox.instance(), "");
  });
});
