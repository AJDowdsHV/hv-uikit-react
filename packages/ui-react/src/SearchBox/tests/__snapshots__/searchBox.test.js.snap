// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> renders correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <Search16 />
</div>
`;
